src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -299,9 +299,9 @@ static inline int MPIDI_OFI_am_isend_long(int rank,
          * is not available here */
         lmt_info->rma_key =
             MPIDI_OFI_index_allocator_alloc(MPIDI_Global.mr_key_allocator, MPL_MEM_RMA);
-        MPIR_ERR_CHKANDSTMT(lmt_info->rma_key >=
-                            (((uint64_t) 1) << (MPIDI_Global.max_mr_key_size * 8)), mpi_errno,
-                            MPI_ERR_OTHER, goto fn_fail, "**ofid_mr_reg");
+        MPIR_ERR_CHKANDSTMT(!MPIDI_OFI_index_is_valid(lmt_info->rma_key,
```

Comment:
Same comment about passing global var through the API.

Reply:
Answered above.